### PR TITLE
chore(CI): test on GHA macOS-14 runner

### PR DESCRIPTION
**Pull request by mayeut**
PyPA is now on an enterprise plan.
Let's use the new macos-13-xlarge runner in order to test on macOS arm64.

### CODE REVIEW

**Comment by mayeut** [comment]
Seems macos-13-xlarge is only available as a paid option (Included minutes cannot be used for larger runners.)
It might be worth discussing this with @pradyunsg who'll have a chat early next year with GitHub folks.

**Comment by joerick** [comment]
@pradyunsg, quick note from us at cibuildwheel, if you do have another discussion with Github about support of PyPA, it would be great if we could get access to 'larger' runners under the plan. 

Testing on macOS arm64 is pretty spotty currently, we only have CirrusCI that supports it and we can't afford to run that on PRs, only on main, as otherwise we run out of quota every month.

Github [recently released hosted runners for macOS arm64](https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/), but they're of the 'larger' type, which are priced per-minute and seemingly not included in the current plan.

So access to these runners would be great, if at all possible!

**Comment by henryiii** [comment]
Looks like M1's are now available generally:

https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

File: .github/workflows/test.yml
Comment (henryiii) [inline .github/workflows/test.yml:41]:
```suggestion
        os: [ubuntu-20.04, windows-latest, macos-11, macos-14]
```

**Comment by henryiii** [comment]
We can rebase on #1743 once it's in, no need to merge this PR before a patch release, unless you want to.

**Comment by joerick** [comment]
Lets get the examples and docs updated first. We might as well provide a config that does native compilation on both x86_64 and arm64 while both runners are available IMO.

File: examples/github-minimal.yml
Comment (joerick) [inline examples/github-minimal.yml:11]: I do wonder if it looks a bit like a mistake to list both these versions here. Worth a comment?

Comment (henryiii) [reply examples/github-minimal.yml:11]: I think I would, since you can't read from this that one is Intel and one ARM. Why not `macOS-13`, though?

File: docs/faq.md
Comment (lelit) [inline docs/faq.md:84]: s/be be/can be/

Comment (mayeut) [inline docs/faq.md:84]:
```suggestion
- On an x86_64 runner, arm64 code can be compiled but it can't be tested. `cibuildwheel` will raise a warning to notify you of this - these warnings can be silenced by skipping testing on these platforms: `CIBW_TEST_SKIP: "*_arm64 *_universal2:arm64"`.
```

Comment (joerick) [reply docs/faq.md:84]: good spot! this took me 30 seconds to actually find 😅

File: examples/github-minimal.yml
Comment (joerick) [reply examples/github-minimal.yml]: have changed it to macos-13, along with our tests

File: docs/faq.md
Comment (radarhere) [inline docs/faq.md:83]:
```suggestion
- On an arm64 runner, it is possible to test x86_64 wheels and both parts of a universal2 wheel using Rosetta 2 emulation.
```

**Review by joerick — APPROVED** [review]

**Pull request merged**
